plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue

```diff
@@ -595,6 +625,119 @@ export default defineComponent({
 
       return true;
     },
+    movePeriod(direction: number) {
+      if (!this.canMovePeriod(direction)) {
+        return;
+      }
+
+      let newDate = new Date();
+      if (this.dateValue != null) {
+        newDate = this.dateValue;
+      }
+
+      switch (this.periodValue) {
+        case 'day':
+          newDate.setDate(newDate.getDate() + direction);
+          break;
+        case 'week':
+          newDate.setDate(newDate.getDate() + direction * 7);
+          break;
+        case 'month':
+          newDate.setMonth(newDate.getMonth() + direction);
+          break;
+        case 'year':
+          newDate.setFullYear(newDate.getFullYear() + direction);
+          break;
+        default:
+          break;
+      }
+
+      // Ensure the date is not outside the min and max dates
+      if (this.dateValue! < piwikMinDate) {
+        this.dateValue = piwikMinDate;
+      }
+      if (this.dateValue! > piwikMaxDate) {
+        this.dateValue = piwikMaxDate;
+      }
+
+      this.onApplyClicked();
+    },
+    isPeriodMoveDisabled(direction: number) {
+      // disable period move when date range is used or when we would go out of the min/max dates
+      if (this.dateValue === null) {
+        return this.isRangeSelection;
+      }
+      return this.isRangeSelection || !this.canMovePeriod(direction);
+    },
+    canMovePeriod(direction: number) {
+      if (this.dateValue === null) {
+        return false;
+      }
+      // atBoundary means we are on the current day, week, month or year
+      // and another move would take us before the site was created or into the future.
+      let atBoundary = false;
+
+      const year = this.dateValue!.getFullYear();
+      const month = this.dateValue!.getMonth();
+      const day = this.dateValue!.getDate();
+      const week = this.getWeekNumber(this.dateValue!);
+
+      const boundaryDate = (direction === -1) ? piwikMinDate : piwikMaxDate;
+
+      switch (this.periodValue) {
+        case 'day':
+          atBoundary = year === boundaryDate.getFullYear()
+                    && month === boundaryDate.getMonth()
+                    && day === boundaryDate.getDate();
+          break;
+        case 'week':
+          atBoundary = year === boundaryDate.getFullYear()
+                    && week === this.getWeekNumber(boundaryDate);
+          break;
+        case 'month':
+          atBoundary = year === boundaryDate.getFullYear()
+                    && month === boundaryDate.getMonth();
+          break;
+        case 'year':
+          atBoundary = year === boundaryDate.getFullYear();
+          break;
+        default:
+          break;
+      }
+
+      return !atBoundary;
+    },
+    getWeekNumber(date: Date) {
```

Comment:
Should we maybe move those new methods to `plugins/CoreHome/vue/src/Periods/utilities.ts` for reusability?

Reply:
Yep, can do that. I came across the Periods when I was looking if the methods existed somewhere already but then left it in the selector component. Will move it.